### PR TITLE
Support Windows-style line endings

### LINKED ISSUE #22: Adds windows-style line endings to unix style file
In Windows, when running docformatter on a file with unix-style line endings, it replaces all line endings with Windows-style endings. I assume it picks the type of line endings based on the OS, rather than the file, which is a pretty bad assumption. Other tools like autopep8 and yapf don't seem to have this issue.

### PR DESCRIPTION
**Pull request by serhiy-yevtushenko**
This closes #22.

Please review, and if accepted, create new release of docformatter( This issue is affecting me, as it prevents using docformatter in pre-commit hooks - leads to changes in file only due to line endings, and conflicts with reorder_python_imports)

### CODE REVIEW

File: docformatter.py [hunk 1]
```diff
@@ -380,6 +386,39 @@ def _find_shortest_indentation(lines):
...
+    Input is a list of lines.
+    """
+    assert not isinstance(source, unicode)
+
+    counter = collections.defaultdict(int)
+    for line in source:
+        if line.endswith(CRLF):
+            counter[CRLF] += 1
+        elif line.endswith(CR):
+            counter[CR] += 1
```
Comment (myint) [inline docformatter.py]: Add a unit test for this function that hits this line. Currently, it is not covered by any test.

Note that I already wrote exhaustive tests for this in `autopep8` that you could use.

File: docformatter.py [hunk 2]
```diff
@@ -380,6 +386,39 @@ def _find_shortest_indentation(lines):
 
     return indentation or ''
 
+def find_newline(source):
+    """Return type of newline used in source.
+    Input is a list of lines.
```
Comment (myint) [inline docformatter.py]: Please run `docformatter` on this code. This does not follow PEP 257. 😄

File: docformatter.py [hunk 3]
```diff
@@ -380,6 +386,39 @@ def _find_shortest_indentation(lines):
...
+        elif line.endswith(CR):
+            counter[CR] += 1
+        elif line.endswith(LF):
+            counter[LF] += 1
+    return (sorted(counter, key=counter.get, reverse=True) or [LF])[0]
+
+
+def normalize_line(line, newline):
+    """Return line with fixed ending, if ending was present in line.
+       Otherwise, does nothing.
```
Comment (myint) [inline docformatter.py]: Similar PEP 257 violation that I mentioned elsewhere.

File: docformatter.py [hunk 4]
```diff
@@ -380,6 +386,39 @@ def _find_shortest_indentation(lines):
...
+    """
+    stripped = line.rstrip('\n\r')
+    if stripped != line:
+        return stripped + newline
+    return line
+
+
+def normalize_line_endings(lines, newline):
+    """Return fixed line endings.
+    All lines will be modified to use the most common line ending.
```
Comment (myint) [inline docformatter.py]: Similar PEP 257 violation that I mentioned elsewhere.

File: docformatter.py [hunk 5]
```diff
@@ -53,6 +53,9 @@
 
 HEURISTIC_MIN_LIST_ASPECT_RATIO = .4
 
+CR = '\r'
+LF = '\n'
+CRLF = '\r\n'
```
Comment (myint) [inline docformatter.py]: There should be an empty line after this to conform to PEP 8.

File: test_docformatter.py [hunk 6]
```diff
@@ -822,6 +822,25 @@ def test_format_code_with_syntax_error(self):
         self.assertEqual('"""\n',
                          docformatter.format_code('"""\n'))
 
+    def test_format_code_dominant_line_ending_style_preserved(self):
```
Comment (myint) [inline test_docformatter.py]: Please also add a separate test for '\r' line endings. See my unit tests in `autopep8`.

**Comment by serhiy-yevtushenko** [comment]
The changes requested were added to code

**Pull request merged**